lib/ansible/plugins/loader.py

```diff
@@ -605,11 +604,17 @@ def _load_plugin_filter():
         if version == u'1.0':
             # Modules and action plugins share the same blacklist since the difference between the
             # two isn't visible to the users
-            filters['ansible.modules'] = frozenset(filter_data['module_blacklist'])
-            filters['ansible.plugins.action'] = filters['ansible.modules']
+            try:
+                filters['ansible.modules'] = frozenset(filter_data['module_blacklist'])
```

Comment:
Sine you're catching TypeError, it looks like only the above line could be causing it.  So it should be the only thing in the try: except.